quantum/process_keycode/process_auto_shift.c

```diff
@@ -392,8 +392,17 @@ bool process_auto_shift(uint16_t keycode, keyrecord_t *record) {
                 break;
 #    endif
         }
-        // If Retro Shift is disabled, possible custom actions shouldn't happen.
-        // clang-format off
+            // If Retro Shift is disabled, possible custom actions shouldn't happen.
+            // clang-format off
+#   if defined(RETRO_SHIFT) && !defined(NO_ACTION_TAPPING)
+#       if defined(HOLD_ON_OTHER_KEY_PRESS_PER_KEY)
+            const bool is_hold_on_interrupt = get_hold_on_other_key_press(keycode, record);
+#       elif defined(IGNORE_MOD_TAP_INTERRUPT)
+            const bool is_hold_on_interrupt = false;
+#       else
+            const bool is_hold_on_interrupt = IS_MT(keycode);
+#       endif
```

Comment:
When rewriting this, I realised that the original code totally ignores the existence or the possibility of `PERMISSIVE_HOLD`. I'm not sure if that's an intentional omission because it's irrelevant or not.